.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,59 @@
+name: E2E test
+on:
+  pull_request:
+      branches:
+        - master
+  workflow_dispatch:
+    environment:
+      description: 'Environment to run tests against'
+      type: environment
+      required: true 
+env:
+  TEST_USERNAME: admin
+  TEST_PASSWORD: password
+  CATTLE_BOOTSTRAP_PASSWORD: password
+  TEST_BASE_URL: https://127.0.0.1:8005
+  API: https://127.0.0.1
+jobs:
+  test:
+    if: "!contains( github.event.pull_request.labels.*.name, 'ci/skip-e2e')"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+      
+      - name: Install packages
+        run: yarn install
+      
+      - name: Prepare build
+        run: yarn e2e:pre-prod
+      
+      - name: Run tests
+        run: yarn e2e:prod
+
+      - name: Upload screenshots
+        uses: actions/upload-artifact@v2
+        if: ${{ failure() }}
+        with:
+          name: run_${{github.run_number}}_id_${{github.run_id}}_screenshots
+          path: cypress/screenshots
+
+      # Disabled due freezing issues related to the CI machine resources
```

Comment:
It's more important to get the PR in, so this is ok, but we should create an issue to re-enable. Having a video is really important to debugging failed e2e tests. 

The epinio QA team have managed to get these working, it's worth syncing with them https://github.com/epinio/epinio-end-to-end-tests/blob/main/.github/workflows/master_rancher_ui_workflow.yml#L131

Update: It looks like the videos are still produced (see `Video` section of each test in https://github.com/rancher/dashboard/runs/6596796767?check_suite_focus=true). Whilst the upload is  disabled can the create of them also be disabled?

Reply:
> Drone was initially failing with a 524 error when trying to download cypress. I've kicked it off again but there's now a lint failure

It was randomly failing before for other reasons, so I stopped bothering, oops.
I can see only 1 linting error though 🤔 
Now it's fixed.

Unfortunately we have no automation to lint code while committing. We should integrate Husky for this task. The [tech debt is written since 3 months](https://confluence.suse.com/display/CU/Health+of+Codebase) but it has not been discussed.

> Still lots of unresolved conversations

I've found only 1 conversation unresolved, but should be you the one to resolve them, no?

> The Epinio QA team have managed to get these working, it's worth syncing with them

They use a standalone app which definitely does not suck all the RAM and CPU like Rancher. What do you suggest me to ask them?

The link point to `retention-days: 7`, do you want to set a limitation for holding the artifacts? I am not so sure 7 days are actually enough, sometimes the PR hangs for weeks.

Already created while removing it. https://github.com/rancher/dashboard/issues/6048
Adding a comment in the description

> can the create of them also be disabled?

No there's not an option and is a feature requested in Cypress since many years.